src/esp/physics/bullet/BulletRigidObject.h

```diff
@@ -13,88 +17,381 @@
+   * btRigidBody</A>\::setDamping. Does not affect @ref RigidObjectType::SCENE.
+   * @param linearDamping The new scalar linear damping coefficient of the
+   * object.
+   */
   void setLinearDamping(const double linearDamping);
+
+  /** @brief Set the scalar angular damping coefficient for the object.
+   * See <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtRigidBody.html">
+   * btRigidBody</A>\::setDamping. Does not affect @ref RigidObjectType::SCENE.
+   * @param angularDamping The new scalar angular damping coefficient for the
+   * object.
+   */
   void setAngularDamping(const double angularDamping);
-  //! Set Margin only works for objects
+
+  /** @brief Set the scalar collision margin of an object. Does not affect @ref
+   * RigidObjectType::SCENE. See <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtCompoundShape.html">
+   * btCompoundShape</A>\::setMargin.
+   * @param margin The new scalar collision margin of the object.
+   */
   void setMargin(const double margin);
 
  protected:
-  //! Needed after changing the pose from Magnum side
-  //! Not exposed to end user
+  /** @brief Used to synchronize Bullet's notion of the object state
+   * after it was changed kinematically. Called automatically on kinematic
+   * updates. See <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtRigidBody.html">
+   * btRigidBody</A>\::setWorldTransform. */
   void syncPose();
 
  private:
-  //! One object can only exist in one world
+  /** @brief A pointer to the Bullet world to which this object belongs. See <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtDiscreteDynamicsWorld.html">
+   * btDiscreteDynamicsWorld</A> .*/
   std::shared_ptr<btDiscreteDynamicsWorld> bWorld_;
 
-  //! Physical scene
-  //! Scene data: triangular mesh shape
-  //! All components are stored as a vector of bCollisionBody_
+  // === Physical scene ===
+
+  //! Scene data: Bullet triangular mesh vertices
   std::unique_ptr<btTriangleIndexVertexArray> bSceneArray_;
+
+  //! Scene data: Bullet triangular mesh shape
   std::vector<std::unique_ptr<btBvhTriangleMeshShape>> bSceneShapes_;
+
+  /** @brief Scene data: All components of a @ref RigidObjectType::SCENE are
+   * stored here. See <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtCollisionObject.html">
+   * btCollisionObject</A>
+   */
   std::vector<std::unique_ptr<btCollisionObject>> bSceneCollisionObjects_;
 
-  // Physical object
+  // === Physical object ===
+
   //! Object data: Composite convex collision shape
-  //! All components are wrapped into one rigidBody_
   std::vector<std::unique_ptr<btConvexHullShape>> bObjectConvexShapes_;
+
+  //! Object data: All components of the collision shape
   std::unique_ptr<btCompoundShape> bObjectShape_;
+
+  /** @brief Object data: All components of a @ref RigidObjectType::OBJECT are
+   * wrapped into one <A
+   * HREF="https://pybullet.org/Bullet/BulletFull/classbtRigidBody.html">
+   * btRigidBody</A>
```

Comment:
~~You can use Markdown links here, these look a bit less ugly -- `[btRigidBody](https://pybullet.org/Bullet/BulletFull/classbtRigidBody.html)`~~

Actually, I think we could hook up a Doxygen tagfile for all these, similarly how it's done for STL and Magnum type crosslinking. I'll take a look at that. (Sorry that you had to manually add all these.)

Reply:
Copy paste does get old, haha. If you have an optimization I'm happy to apply it!

Reply:
I looked around and Bullet doesn't seem to provide any tagfile on its own, so I would need to generate one myself from the sources. Gonna do that tomorrow, it's getting late here :)

Reply:
Done: #222. Have fun :)

Reply:
Thanks again, mosra! I'll port these as soon as #222 is live.